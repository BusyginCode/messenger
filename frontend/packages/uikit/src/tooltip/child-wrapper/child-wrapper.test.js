import React from 'react'
import { shallow } from 'enzyme'
import ChildWrapper from './child-wrapper'

describe('uikit/Tooltip/ChildWrapper', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <ChildWrapper>
        <button>Press</button>
      </ChildWrapper>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('componentWillUnmount should be called on unmount', () => {
    const wrapper = shallow(
      <ChildWrapper>
        <button>Press</button>
      </ChildWrapper>
    )
    const componentWillUnmount = jest.spyOn(wrapper.instance(), 'componentWillUnmount')
    const onMouseLeave = jest.spyOn(wrapper.instance(), 'onMouseLeave')

    wrapper.unmount()
    expect(componentWillUnmount).toHaveBeenCalled()
    expect(onMouseLeave).toHaveBeenCalled()
  })
})
