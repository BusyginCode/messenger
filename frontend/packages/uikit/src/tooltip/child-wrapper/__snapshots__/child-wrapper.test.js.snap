// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/Tooltip/ChildWrapper should render correctly 1`] = `
<span>
  <button>
    Press
  </button>
</span>
`;
